extensions/wikia/AdEngine/js/lookup/prebid/adapters/rubicon.js

```diff
@@ -27,12 +28,48 @@ define('ext.wikia.adEngine.lookup.prebid.adapters.rubicon', [
 			}
 		};
 
+	function getAdContext() {
+		return adContext.getContext();
+	}
+
+	function getTargeting(slotName, skin) {
+		var provider = skin === 'oasis' ? 'gpt' : 'mobile',
+			s1 = getAdContext().targeting.wikiIsTop1000 ? adLogicZoneParams.getName() : 'not a top1k wiki';
+
+		return {
+			pos: slotName,
+			src: provider,
+			s0: adLogicZoneParams.getSite(),
+			s1: s1,
+			s2: adLogicZoneParams.getPageType(),
+			lang: adLogicZoneParams.getLanguage()
+		};
+	}
```

Comment:
This method is a copy&past from `rubiconDisplay.js`. Can we extract it?

Reply:
Good idea. Refactored.